doc/rule-dsl.md

```diff
@@ -0,0 +1,510 @@
+
+The following destination policy specifies that RANDOM load balancing be used for balancing across
+instances of any version of the "reviews" microservice:
+
+```yaml
+destination: reviews.default.svc.cluster.local
+load_balancing: RANDOM
+```
+
+#### Property: circuit_breaker <a id="circuit_breaker"></a>
+
+The `circuit_breaker` field can be used to set a circuit breaker for a particular microservice. 
+A simple circuit breaker can be set based on a number of criteria such as connection and request limits.
+
+For example, the following destination policy
+sets a limit of 100 connections to "reviews" service version "v1" backends.
+
+```yaml
+destination: reviews.default.svc.cluster.local
+tags:
+  version: v1
+circuit_breaker:
+  simple_cb:
+    max_connections: 100
+```
+
+The complete set of simple circuit breaker fields can be found
+[here](https://github.com/rshriram/api/blob/master/proxy/v1/config/cfg.md#circuitbreakersimplecircuitbreakerpolicy).
+
+### Destination Policy Evaluation <a id="destination-policy-evaluation"></a>
+
+Similar to route rules, destination policies are associated with a particular `destination` however
+if they also include `tags` their activation depends on route rule evaluation results.
+
+The first step in the rule evaluation process evaluates the route rules for a `destination`,
+if any are defined, to determine the tags (i.e., specific version) of the destination service
+that the current request will be routed to. Next, the set of destination policies, if any, are evaluated
+to determine if they apply.
+
+One subtlety of the algorithm to keep in mind is that policies that are defined for specific
+tagged destinations will only be applied if the corresponding tagged instances are explicity
+routed to. For example, consider the following rule, as the one and only rule defined for
+the "reviews" microservice.
+
+```yaml
+destination: reviews.default.svc.cluster.local
+tags:
+  version: v1
+circuit_breaker:
+  simple_cb:
+    max_connections: 100
+```
+
+Since there is no specific route rule defined for the "reviews" microservice, default
+round-robin routing behavior will apply, which will persumably call "v1" instances on occasion,
+maybe even always if "v1" is the only running version. Nevertheless, the above policy will never
+be invoked since the default routing is done at a lower level. The rule evaluation engine will be
+unaware of the final destination and therefore unable to match the destination policy to the request. 
+
+You can fix the above example in one of two ways. You can either remove the `tags:`
+from the rule, if "v1" is the only instance anyway, or, better yet, define proper route rules
+for the service. For example, you can add a simple route rule for "reviews:v1".
+
+```yaml
+destination: reviews.default.svc.cluster.local
+route:
+- tags:
+    version: v1
+```
+
+Setting a default rule like this for every microservice is generally considered a best practice in Istio.
```

Comment:
@rshriram It's probably worth mentioning the implicit default routing (all versions to all versions). That is by default, you can send traffic from any service to any other service.